### PR TITLE
[13.0][MIG] stock_request_analytic

### PR DESCRIPTION
**Pull request by ps-tubtim**
Standard migration.

### CODE REVIEW

**Review by LoisRForgeFlow — APPROVED** [review]
Code review LGTM :+1: 

Thanks for the rebranding :)

File: stock_request_analytic/views/analytic_views.xml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        <field name="inherit_id" ref="analytic.view_account_analytic_account_form" />
+        <field name="arch" type="xml">
+            <xpath expr="//group[@name='main']" position='after'>
+                <notebook groups="stock_request.group_stock_request_user">
+                    <page string="Stock Request" name="stock_request">
+                        <field name="stock_request_ids" />
+                    </page>
+                </notebook>
+            </xpath>
+        </field>
```
Comment (LoisRForgeFlow) [inline stock_request_analytic/views/analytic_views.xml]: One suggestion, could you move this to a smart button? this list can get huge quite quckly

Comment (ps-tubtim) [reply stock_request_analytic/views/analytic_views.xml]: Thank you for your suggestion. I already update this code. Could you review it again?

File: stock_request_analytic/models/stock_request.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+            and r.analytic_account_id
+            and r.analytic_account_id.company_id != r.company_id
+            for r in self
+        ):
+            raise ValidationError(
+                _(
+                    "You cannot link a analytic account to a stock request "
+                    "that belongs to another company."
+                )
+            )
```
Comment (joao-p-marques) [inline stock_request_analytic/models/stock_request.py]: Could this be replaced by adding `check_company=True` to the `analytic_account_id` field?
```suggestion
class StockRequest(models.Model):
    _inherit = "stock.request"
    _check_company_auto = True

    analytic_account_id = fields.Many2one(
        "account.analytic.account", string="Analytic Account", check_company=True
    )
```

File: stock_request_analytic/tests/test_stock_request_analytic.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    def test_stock_multi_analytic(self):
+        vals = self.prepare_order_request_multi_analytic(
+            self.analytic, self.analytic3, self.main_company
+        )
+        order = self.env["stock.request.order"].create(vals)
+        order.action_confirm()
+        self.assertEqual(order.analytic_count, 2)
+
+    def test_company(self):
+        with self.assertRaises(ValidationError):
```
Comment (joao-p-marques) [inline stock_request_analytic/tests/test_stock_request_analytic.py]: Tests need to be adapted if the suggestion above is included
```suggestion
        with self.assertRaises(UserError):
```

**Review by joao-p-marques — COMMENTED** [review]
Code LGTM :+1: 

Just a small suggestion.

Also, could you include https://github.com/OCA/stock-logistics-warehouse/pull/1105 and squash the bot commits?

File: stock_request_analytic/models/stock_request.py [hunk 4]
```diff
@@ -1,30 +1,20 @@
...
+# Copyright 2017-2020 ForgeFlow, S.L. (https://www.forgeflow.com)
 # License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
 
-from odoo import _, api, fields, models
-from odoo.exceptions import ValidationError
+from odoo import fields, models
 
 
 class StockRequest(models.Model):
     _inherit = "stock.request"
```
Comment (joao-p-marques) [inline stock_request_analytic/models/stock_request.py:8]: I think you need to add `_check_company_auto = True` here for the check to work: https://www.odoo.com/documentation/13.0/howtos/company.html?highlight=multi%20company#multi-company-consistency

File: stock_request_analytic/models/stock_request.py [hunk 5]
```diff
@@ -1,30 +1,20 @@
...
 
     analytic_account_id = fields.Many2one(
-        "account.analytic.account", string="Analytic Account"
+        "account.analytic.account",
+        string="Analytic Account",
+        check_company=True,
+        domain="['|', ('company_id', '=', False), ('company_id', '=', company_id)]",
```
Comment (joao-p-marques) [inline stock_request_analytic/models/stock_request.py:14]: I think the suggestion above would remove the need for this domain

File: stock_request_analytic/models/stock_request.py [hunk 6]
```diff
@@ -1,30 +1,20 @@
...
+    )
+    analytic_tag_ids = fields.Many2many(
+        "account.analytic.tag",
+        string="Analytic Tags",
+        domain="['|', ('company_id', '=', False), ('company_id', '=', company_id)]",
```
Comment (joao-p-marques) [inline stock_request_analytic/models/stock_request.py:19]: The same here as above

**Review by joao-p-marques — COMMENTED** [review]
Thanks for the changes @ps-tubtim!

Can you take a look at the rest of the comments and to the other Travis error please?

@pedrobaeza can you also help reviewing?

**Comment by pedrobaeza** [comment]
Superseded by #1124

**Pull request closed without merging**